undistort_test.py

```diff
@@ -0,0 +1,95 @@
+import open3d as o3d
+import argparse
+import numpy as np
+import mrob
+import math
+
+from pathlib import Path
+from scipy.spatial.transform import Rotation as R
+from collections import defaultdict
+
+
+def get_intermediate_transform(initial_pose, target_pose, coef):
+    dxi = mrob.geometry.SE3(target_pose @ np.linalg.inv(initial_pose)).Ln()
+    return mrob.geometry.SE3(coef * dxi).T() @ initial_pose
+
+def undistort_segment(pcd, theta_min, theta_max, color, transform):
+    # Convert the point cloud points to a numpy array
+    points = np.asarray(pcd.points)
+    thetas = np.asarray([math.atan2(p[1],p[0]) for p in points])
+
+    # Create a mask for points that fall within the ROI
+    mask_roi = (thetas < theta_max) & (thetas >= theta_min)
+    trues = mask_roi.sum()
+
+    new_pcd = o3d.geometry.PointCloud()
+    if trues > 0:
+        new_colors = np.full((trues, 3), color)
+        new_points = points[mask_roi]
+
+        new_pcd.points = o3d.utility.Vector3dVector(new_points)
+        new_pcd.colors = o3d.utility.Vector3dVector(new_colors)
+
+    return new_pcd.transform(transform)
```

Comment:
Looks like transform is an unnecessary parameter for this function. You can .transform it in `undistort_cloud`